src/pluginutils.c

```diff
@@ -554,6 +554,24 @@ void plugin_builder_connect_signals(GeanyPlugin *plugin,
 }
 
 
+/** Get the additional data that corresponds to the plugin.
+ *
+ * @param plugin The plugin provided by Geany
+ * @return The data corresponding to the plugin or @c NULL if none set.
+ *
+ * @since 1.29 (API 230)
+ *
+ * @see geany_plugin_set_data()
+ */
+gpointer geany_plugin_get_data(const GeanyPlugin *plugin)
+{
+	g_return_val_if_fail (plugin != NULL, NULL);
+	g_return_val_if_fail (PLUGIN_LOADED_OK (plugin->priv), NULL);
```

Comment:
Look at [the function below](https://github.com/geany/geany/pull/1234/commits/84720babfbce2642289a8c038137402533e5d2c5#diff-68052c9feb0f8c0562bf6f40b8b19669R602) this one. I copied the pre-conditions but added an additional one for the `plugin != NULL` for good measure.